pipeline/join.go

```diff
@@ -90,11 +91,23 @@ func newJoinNode(e EdgeType, parents []Node) *JoinNode {
 }
 
 // Prefix names for all fields from the respective nodes.
+//
 // Each field from the parent nodes will be prefixed with the provided name and a '.'.
 // See the example above.
 //
 // The names cannot have a dot '.' character.
 //
+// If a prefix is not specified or is empty, then field names from parent nodes
+// will be copied without attaching a prefix.
+//
+// If a prefix contains a trailing '#' character, then the prefix
+// upto, but not including the trailing '#' character is prepended to the field name
+// from the parent node without adding the '.' character after the prefix.
+//
+// It is the callers responsibility to ensure that when these rules are applied
+// the collection of output field names does not contain any duplicate names. Failure
+// to ensure this will result in a runtime error.
```

Comment:
I believe that the task should be stopped since data it is likely that if a set of points fails the uniqueness test then all subsequent sets of points will also fail the test in which case joined data will be silently dropped from the output stream.